Set-CARMLFoldersForPBRSingular.ps1

```diff
@@ -0,0 +1,420 @@
+﻿<#
+.SYNOPSIS
+Re-format the name of every CARML module to match the requirements of the Public Bicep Registry
+
+.DESCRIPTION
+The names are formatted as per the following rules:
+- Add a '-' prefix to every upper-case character
+- Make the entire name lower case
+
+For example, it re-formats
+- 'AppConfiguration\configurationStores\keyValues\deploy.bicep'
+to
+- app-configuration\configuration-stores\key-values\deploy.bicep
+
+.PARAMETER ModulesFolderPath
+Mandatory. The path to the modules who's folder names should be converted.
+
+.EXAMPLE
+Set-CARMLFoldersForPBRSingular -ModulesFolderPath 'C:\dev\ip\Azure-ResourceModules\ResourceModules\modules'
+
+Convert all folders in path 'C:\dev\ip\Azure-ResourceModules\ResourceModules\modules'
+
+.EXAMPLE
+Set-CARMLFoldersForPBRSingular -ModulesFolderPath 'C:\dev\ip\GitHub\Azure\ResourceModules\modules' -WorkflowsPath 'C:\dev\ip\GitHub\Azure\ResourceModules\.github\workflows' -PipelinesPath 'C:\dev\ip\GitHub\Azure\ResourceModules\.azuredevops\modulePipelines' -UtilitiesPath 'C:\dev\ip\GitHub\Azure\ResourceModules\utilities' -Verbose
+
+Convert all folders in path 'C:\dev\ip\Azure-ResourceModules\ResourceModules\modules' and replaces all pipeline, workflows and utility references
+
+.EXAMPLE
+Set-CARMLFoldersForPBRSingular -ResourceProviderPath 'C:\dev\ip\Azure-ResourceModules\ResourceModules\modules\KeyVault'
+
+Convert all folders in path 'C:\dev\ip\Azure-ResourceModules\ResourceModules\modules\KeyVault' included
+
+.NOTES
+This script should only be run AFTER the 'Microsoft.' provider namespace prefix was removed.
+Identifiers such as 'DBforPostgreSQL' becomes 'd-bfor-postgre-s-q-l' which should probably be manually reverted to 'db-for-postgre-sql'.
+#>
+function Set-CARMLFoldersForPBRSingular {
+
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $ModulesFolderPath,
+
+        [Parameter(Mandatory = $false)]
+        [string] $ResourceProviderPath = '',
+
+        [Parameter(Mandatory = $false)]
+        [string] $WorkflowsPath = '',
+
+        [Parameter(Mandatory = $false)]
+        [string] $PipelinesPath = '',
+
+        [Parameter(Mandatory = $false)]
+        [string] $UtilitiesPath = ''
+    )
+
+    $specialConversionHash = @{
```

Comment:
Why did you not use the `rtMapping.json`file we created for this purpose? :D

Reply:
I just copied there the content as the script was designed that way. I can retrieve the json content instead so that we don't have duplicated content. Anyway please consider the PR draft, I'm testing on the *_readme one and will open to review both PRs only when all tests are green

Reply:
leveraging json

Reply:
Ready for review